internal/provider/provider.go

```diff
@@ -337,3 +358,91 @@ func (p *btpcliProvider) DataSources(ctx context.Context) []func() datasource.Da
 		newWhoamiDataSource,
 	}, betaDataSources...)
 }
+
+func determineAuthFlow(config providerData, idToken string) string {
+	if len(idToken) > 0 {
+		return idTokenFlow
+	} else if !config.TLSClientKey.IsNull() && !config.TLSClientCertificate.IsNull() {
+		return x509Flow
+	} else {
+		return userPasswordFlow
+	}
+}
+
+func validateGlobalAccount(globalAccount string, resp *provider.ConfigureResponse) {
```

Comment:
This is probably better checked using a stringvalidator directly in the schema? Maybe we can even check all of the other constraints using stringvalidator.ConflictsWith() and stringvalidator.AlsoRequires()?

Reply:
We already have several constraints in the schema. However, the output is not very consumer friendly. That's why we wanted to add the error messages in the provider to give a more human readable error message. 

For the global account I can transfer the check to a string validator. For the rest and their not easy to digest combinations I would stick to the error messages.